postgresql/cluster/bdr/slave.sls

```diff
@@ -0,0 +1,33 @@
+{%- from "postgresql/map.jinja" import server with context %}
+{%- from "postgresql/map.jinja" import cluster with context %}
+{%- if cluster.enabled %}
+
+{%- set repuser = cluster.user.get('repuser') %}
+
+postgresql_user_cluster_repuser:
+  postgres_user.present:
+  - name: repuser
+  - user: postgres
+  - replication: True
+  - superuser: True
+  - password: {{ repuser.password }}
+  - require:
+    - service: postgresql_service
+    - pkg: postgresql_packages
+
+{%- set nodename = salt.grains.get('nodename') %}
+
+{%- for database_name, database in server.get('database', {}).iteritems() %}
+bdr_group_create_{{ database_name }}:
+  module.run:
+    - name: postgres.psql_query
+    - query: SELECT bdr.bdr_group_join(local_node_name := '{{ nodename }}', node_external_dsn := 'host={{ cluster.get("local") }} user=repuser password={{ repuser.password }} dbname={{ database_name }}', join_using_dsn := 'host={{ cluster.get("master") }} user=repuser password={{ repuser.password }} dbname={{ database_name }}');
+    - user: postgres
+    - maintenance_db: {{ database_name }}
+    - unless: sudo -H -u postgres bash -c 'timeout 10s psql {{ database_name }} -c "SELECT bdr.bdr_node_join_wait_for_ready();"'
```

Comment:
I tested this behaviour and it seems salt executes unless test scripts as root, because it only works with sudo -H -u postgres